hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        diff = self[xray_lines[0]].centre.value - ref[0]
+        offset_old = self.spectrum.axes_manager[-1].offset
+        self.spectrum.axes_manager[-1].offset -= diff
+        offset = self.spectrum.axes_manager[-1].offset
+        print "Offset changed from  %lf to %lf" % (offset_old, offset)
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            component.centre.value = ref[i]
+
+    def calibrate_energy_axis(self,
+                              calibrate='resolution',
+                              xray_lines='all_alpha',
+                              spread_to_all_lines=True,
+                              **kwargs):
+        """
+        Calibrate the resolution, the scale or the offset of the energy axis
+        by fitting.
+
+        Parameters
+        ----------
+        calibrate: 'resolution' or 'scale' or 'offset'
+            If 'resolution', fits the width of Gaussians place at all x-ray
+            lines. The width is given by a model of the detector resolution,
+            obtained by extrapolation the `energy_resolution_MnKa` in `metadata`
+            This method will update the value of `energy_resolution_MnKa`.
+            If 'scale', calibrate the scale of the energy axis
+            If 'offset', calibrate the offset of the energy axis
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        **kwargs : extra key word arguments
+            All extra key word arguments are passed to fit or
+            multifit, depending on the value of kind.
+
+        """
+
+        if calibrate == 'resolution':
+            free = self._twin_xray_lines_width
+            fix = self.fix_xray_lines_width
+            scale = self._set_energy_resolution
+        elif calibrate == 'scale':
+            free = self._twin_xray_lines_scale
+            fix = self.fix_xray_lines_energy
+            scale = self._set_energy_scale
+        elif calibrate == 'offset':
+            free = self._twin_xray_lines_offset
+            fix = self.fix_xray_lines_energy
+            scale = self._set_energy_offset
+        ref = free(xray_lines=xray_lines)
+        self.fit(**kwargs)
+        fix(xray_lines=xray_lines)
+        scale(xray_lines=xray_lines, ref=ref)
+        self.update_plot()
+
+    def free_sub_xray_lines_weight(self, xray_lines='all', bound=0.01):
+        """
+        Free the weight of a sub X-ray lines
+
+        Remove the twin on the height of sub-Xray lines (non alpha)
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all'
+            The Xray lines. If 'all', fit all lines
+        bounds: float
+            Bound the height of the peak to a fraction of
+            its height
+        """
+        def free_twin():
```

Comment:
It's not strictly necessary, but I would add an argument `component` to this function. Otherwise the usage makes me cringe